apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -921,9 +922,16 @@ mod test_utils {
         /// Add a wrapper tx to the queue of txs to be decrypted
         /// in the current block proposal
         #[cfg(test)]
-        pub fn enqueue_tx(&mut self, wrapper: WrapperTx) {
+        pub fn enqueue_tx(
+            &mut self,
+            wrapper: WrapperTx,
+            inner_tx: Option<EncryptedTx>,
```

Comment:
It is a leak of information here to indicate whether the inner tx contains data or code or not.

Reply:
If you're talking more generally about the `inner_tx: Option<EncryptedTx>` field of `Tx`, then I believe this isn't necessarily the case... For a valid transaction, the presence of a `WrapperTx` (with its mandatory `tx_hash: Hash` field) implies that the `inner_tx` field of its container is necessarily occupied by `Some(...)`. Hence for the current kinds of transactions we have, I think this information would already be known/be derivable.

The reason why `inner_tx: Option<EncryptedTx>` was given an option type was to allow hardware-constrained hardware wallets to receive abbreviated/truncated versions of transactions (without making more types) to sign over.

Reply:
I see. Sounds reasonable